docs/plugins/loadext.rst

```diff
@@ -0,0 +1,53 @@
+Load Extension Plugin
+=====================
+
+Beets uses an SQLite database to store and query library information, which
+has support for extensions to extend its functionality.
+
+One of the primary uses of this within beets is with the `"ICU" extension`_,
+which adds support for case insensitive querying of non-ASCII characters.
+
+.. _"ICU" extension: https://www.sqlite.org/src/dir?ci=7461d2e120f21493&name=ext/icu
+
+Configuration
+-------------
+
+To configure the plugin, make a ``loadext`` section in your configuration
+file. The section must consist of a list of paths to extensions to load.
+
+If a relative path is specified, it is resolved relative to the beets
+configuration directory.
+
+No file extension is required, as SQLite will automatically use the correct
+extension for the current platform.
+
+Building the ICU extension
+--------------------------
+This section is for **advanced** users only, and is not an in-depth guide on
+building the extension.
+
+To compile the ICU extension, you will need a few dependencies:
+
+ - gcc
+ - icu-devtools
+ - libicu
+ - libicu-dev
+ - libsqlite3-dev
+
+The following commands can be then be used to compile the ICU extension:
+
+.. code-block:: shell
+
+    $ wget https://sqlite.org/2019/sqlite-src-3280000.zip
+    $ unzip sqlite-src-3280000.zip
+    $ cd sqlite-src-3280000/ext/icu
+    $ gcc -shared -fPIC icu.c `icu-config --ldflags` -o libicu.so
+    $ cp libicu.so [your beets config directory]
```

Comment:
To make this a little more digestible, I'd probably just say:

```
$ cp libicu.so ~/.config/beets
```

Readers (advanced ones, anyway) will be able to sort out that they need to think about where to copy the file.